### DISCUSSION TITLE
my commit isnt showing

### DISCUSSION THREAD

**Discussion by rylena** [community/community#144762] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

i recently made a commit using GIT and not GitHub and i think i configured my user and email incorrectly and due to the commit is not shown in my account how can i fix this?

**Comment by mastertanzeel** [comment, marked as answer]
Hi @rylena ,

To fix this, you’ll need to amend the commit with the correct user name and email. First, configure Git with the correct credentials using the commands: git config user.name "Your Name" and git config user.email "your-email@example.com". Then, use git commit --amend --reset-author to update the commit with the new author information. After amending, force-push the changes to the remote repository using git push --force. This will overwrite the previous commit with the corrected details. Note that if this repository is shared with others, inform them about the force-push to avoid conflicts.

I hope this will help you.

**Comment by SameerNadaf** [comment]
If your commit isn’t showing in your GitHub account because you used the wrong user/email, don’t worry—you can fix it by amending the commit and pushing the updated information to GitHub. Here's a simple step-by-step guide:

1. Check Your Current Configuration
Run these commands to check your current user.name and user.email:

bash
git config user.name
git config user.email
2. Set the Correct Name and Email
Set your correct GitHub username and email (the email you use on your GitHub account):

bash
git config user.name "Your GitHub Username"
git config user.email "your_email@example.com"
3. Amend the Commit
If the wrong commit is the most recent one, you can easily amend it:

bash
git commit --amend --reset-author --no-edit
--reset-author: Updates the commit with the new name and email.
--no-edit: Keeps the existing commit message.
4. Push the Amended Commit
If you already pushed the commit to a remote repository, you’ll need to force-push it with:

bash
git push --force
5. Verify on GitHub
Go to your GitHub repository and check if the commit is now linked to your account.

If the Commit is NOT the Most Recent One
If the incorrect commit is older:

Use interactive rebase to edit the commit:

bash
git rebase -i HEAD~<number_of_commits>
Replace <number_of_commits> with the number of commits to go back.

Follow the interactive instructions to update the commit with the wrong information.

**Comment by CleverLemming1337** [comment]
Did you push your changes using `git push`?
